pymatgen/io/openmm/inputs.py

```diff
@@ -0,0 +1,301 @@
+        with io.StringIO(contents) as s:
+            pdb = PDBFile(s)
+            topology = pdb.getTopology()
+            positions = pdb.getPositions(asNumpy=True)
+        return TopologyInput(topology, positions)
+
+
+class SystemInput(InputFile):
+    def __init__(self, system: System):
+        self.system = system
+
+    def get_string(self) -> str:
+        return XmlSerializer.serialize(self.system)
+
+    @classmethod
+    def from_string(cls, contents: str):
+        return SystemInput(XmlSerializer.deserialize(contents))
+
+
+class IntegratorInput(InputFile):
+    def __init__(self, integrator: Integrator):
+        self.integrator = integrator
+
+    def get_string(self) -> str:
+        return XmlSerializer.serialize(self.integrator)
+
+    @classmethod
+    def from_string(cls, contents: str):
+        return IntegratorInput(XmlSerializer.deserialize(contents))
+
+
+class StateInput(InputFile):
+    def __init__(self, state: State):
+        self.state = state
+
+    def get_string(self) -> str:
+        return XmlSerializer.serialize(self.state)
+
+    @classmethod
+    def from_string(cls, contents: str):
+        return StateInput(XmlSerializer.deserialize(contents))
+
+
+class OpenMMSet(InputSet):
+
+    # TODO: if there is an optional file should it be missing or should the value be none?
+    @classmethod
+    def from_directory(
+        cls,
+        directory: Union[str, Path],
+        topology_file: str = "topology.pdb",
+        system_file: str = "system.xml",
+        integrator_file: str = "integrator.xml",
+        state_file: str = "state.xml",
+    ):
+        topology = TopologyInput.from_file(topology_file)
+        system = TopologyInput.from_file(system_file)
+        integrator = TopologyInput.from_file(integrator_file)
+        openmm_set = OpenMMSet(
+            topology=topology,
+            system=system,
+            integrator=integrator,
+        )
+        if Path("state.xml").is_file():
+            state = TopologyInput.from_file(state_file)
+            openmm_set["state"] = state
+        return openmm_set
+
+    def validate(self) -> bool:
+        # TODO: this should test if the set returns a valid simulation and throw an error if it does not
+        # TODO: is this condition too strict?
+        try:
+            self.get_simulation()
+        except Exception as e:
+            print(
+                "A valid simulation could not be generated, the following error was raised:",
+                e,
+            )
+            return False
+        else:
+            return True
+
+    def get_simulation(self) -> Simulation:
+        simulation = Simulation(
+            self.topology.topology,
+            self.system.system,
+            self.integrator.integrator,
+        )
+        if hasattr(self, "state") and self.state:
+            # TODO: confirm that this works correctly
+            simulation.context.setState(self.state.state)
+        return simulation
```

Comment:
Per comment above and our discussion about integrating `run` into the IO interface, can you add a `run()` method that calls this and actually executes the simulation?

Reply:
I definitely could, but I am not sure we should. 

For other codes, like Packmol, `run()` is a simple operation that would be annoying for the user to execute themself. In OpenMM, once the user has a simulation it's very easy for them to run it with `simulation.run(steps)`. Additionally, a user would likely want to do some equilibration, energy minimization, or annealing before running. If we aren't going to support those, I think a `run` method would only cloud the purpose of the InputSet.

To me, it seems cleaner for pymatgen to handle just IO and to leave running the simulation to the user or to `atomate`.

I'm happy to discuss this more though.

Reply:
I see. I guess this ties in with our broader discussion of where to put customization settings. For example, we _could_ define all those steps as part of the `InputSet` so that `run()` would still make sense (because all the choices have already been made). But that's not to say we should.

I defer to your judgment; it's fine with me to do without `run` until we see how this works in practice and at scale.

Reply:
Agreed, and we can certainly defer a final decision until this IO is more final.

> For example, we could define all those steps as part of the InputSet

While this is true for most IO, this is actually not true in the case of OpenMM. Think of the simulation as the starting point. Once we have a simulation, we can use Python to dynamically execute steps, change the temperature, or change the ensemble, but we don't define those operations in advance. I mean, we could, but we would essentially be dynamically generating a Python script.

That's a key way that OpenMM, and probably other pythonic simulation codes, differ from the traditional approach.